cmake/modules/shields.cmake

```
@@ -79,41 +79,56 @@ foreach(root ${BOARD_ROOT})
 
       list(REMOVE_ITEM SHIELD-NOTFOUND ${s})
 
-      # Add <shield>.overlay to the shield_dts_files output variable.
-      list(APPEND
-        shield_dts_files
-        ${SHIELD_DIR_${s}}/${s}.overlay
-        )
+      # Add <shield>.overlay to a temporary variable
+      set(shield_${s}_dts_file ${SHIELD_DIR_${s}}/${s}.overlay)
 
-      # Add the shield's directory to the SHIELD_DIRS output variable.
-      list(APPEND
-        SHIELD_DIRS
-        ${SHIELD_DIR_${s}}
-        )
+      # Add the shield's directory to a temporary variable
+      set(shield_${s}_dir ${SHIELD_DIR_${s}})
 
       # Search for shield/shield.conf file
       if(EXISTS ${SHIELD_DIR_${s}}/${s}.conf)
-        # Add <shield>.conf to the shield_conf_files output variable.
-        list(APPEND
-          shield_conf_files
-          ${SHIELD_DIR_${s}}/${s}.conf
-          )
+        # Add <shield>.conf to a temporary variable
+        set(shield_${s}_conf_file ${SHIELD_DIR_${s}}/${s}.conf)
       endif()
-
-      # Add board-specific .conf and .overlay files to their
-      # respective output variables.
-      zephyr_file(CONF_FILES ${SHIELD_DIR_${s}}/boards
-                  DTS   shield_dts_files
-                  KCONF shield_conf_files
-      )
-      zephyr_file(CONF_FILES ${SHIELD_DIR_${s}}/boards/${s}
-                  DTS   shield_dts_files
-                  KCONF shield_conf_files
-      )
     endforeach()
   endif()
 endforeach()
 
+# Process shields in-order
+if(DEFINED SHIELD)
+  foreach(s ${SHIELD_AS_LIST})
+    # Add <shield>.overlay to the shield_dts_files output variable.
+    list(APPEND
+      shield_dts_files
+      ${shield_${s}_dts_file}
+      )
+
+    # Add the shield's directory to the SHIELD_DIRS output variable.
+    list(APPEND
+      SHIELD_DIRS
+      ${shield_${s}_dir}
```

I believe we can just use the already shield specific var `SHIELD_DIR_${s}` here and scratch the extra temp var.
```suggestion
      ${SHIELD_DIR_${s}}
```

---

Fixed